core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -184,12 +212,23 @@ object ConsumerPerformance {
                            lastMessagesRead: Long,
                            startMs: Long,
                            endMs: Long,
-                           dateFormat: SimpleDateFormat) = {
+                           dateFormat: SimpleDateFormat,
+                           beginTime: Option[Long] = None,
+                           joinTimeOpt: Option[Long] = None,
+                           periodicJoinTimeInMs: Option[Long] = None) = {
     val elapsedMs: Double = endMs - startMs
     val totalMBRead = (bytesRead * 1.0) / (1024 * 1024)
     val mbRead = ((bytesRead - lastBytesRead) * 1.0) / (1024 * 1024)
-    println("%s, %d, %.4f, %.4f, %d, %.4f".format(dateFormat.format(endMs), id, totalMBRead,
-        1000.0 * (mbRead / elapsedMs), messagesRead, ((messagesRead - lastMessagesRead) / elapsedMs) * 1000.0))
+    print("%s, %d, %.4f, %.4f, %d, %.4f".format(dateFormat.format(endMs), id, totalMBRead,
+      1000.0 * (mbRead / elapsedMs), messagesRead, ((messagesRead - lastMessagesRead) / elapsedMs) * 1000.0))
+    joinTimeOpt match {
```

Comment:
Maybe we should just move this block to a separate function so that we don't need all of the `Option` parameters?

Reply:
Sorry, I am not sure if I fully understand this comment. Do you mean create a new function to handle all the printing things whereas `printProgressMessage` is that type of function?

Reply:
Yes, it's a minor issue, but it's kind of annoying that we have three optional parameters which are all expected in the case of the new consumer. I think a cleaner approach is to use a separate method. Something approximately like this:

```scala
def printBasicProgress ...
def printExtraProgress ...

def printNewConsumerProgress {
  printBasicProgress()
  printExtraProgress()
  println()
}

def printOldConsumerProgress() {
  printBasicProgress()
  println()
}
```
Then we shouldn't need the optional parameters.